Add option to detect bad segments for each channel

HQian96 opened the pull request:
This PR adds the option to detect bad segments from each channel

cgohil8 commented:
Do you actually need to add a new function? Is it not possible to modify `_find_outliers_in_segments` to accept the new parameters you added in `_find_outliers_in_segments_per_channel`?

HQian96 commented:
> Do you actually need to add a new function? Is it not possible to modify `_find_outliers_in_segments` to accept the new parameters you added in `_find_outliers_in_segments_per_channel`?

I have modified `_find_outliers_in_segments`  and added args to allow it do bad segment detection for every channel.
